rendermodules/materialize/delete_materialized_tilesource.py

```diff
@@ -0,0 +1,92 @@
+"""
+delete materializations in a tilesource directory
+"""
+import errno
+import os
+from multiprocessing.pool import ThreadPool
+
+import argschema
+
+from rendermodules.materialize.schemas import (
+    DeleteMaterializedSectionsParameters, DeleteMaterializedSectionsOutput)
+
+example_input = {
+  "minZ": 19750,
+  "maxZ": 19750,
+  "basedir": "/allen/programs/celltypes/production/wijem/workflow_data/production/chunk_30_zs19713_ze20389/17797_1R/FUSEDOUTSTACK/1024x1024",
+  "pool_size": 20
+}
+
+
+class DeleteMaterializedSectionsModule(argschema.ArgSchemaParser):
+    default_schema = DeleteMaterializedSectionsParameters
+    default_output_schema = DeleteMaterializedSectionsOutput
+
+    @staticmethod
+    def get_ts5_tiles(basedir, zs, levels=None):
+        levels = ([int(l) for l in os.listdir(basedir) if l.isdigit()]
+                  if levels is None else levels)
+        for lvl in sorted(levels):
+            lvldir = os.path.join(basedir, str(lvl))
+            for z in zs:
+                zdir = os.path.join(lvldir, str(z))
+                try:
+                    for r in os.listdir(zdir):
+                        rowdir = os.path.join(zdir, r)
+                        # python2 compatible
+                        for cfile_basename in os.listdir(rowdir):
+                            yield os.path.join(rowdir, cfile_basename)
+                except OSError as e:
+                    if e.errno == errno.ENOENT:
```

Comment:
I believe these are dealing with zs that could be missing, since the zs are specified by minZ/maxZ rather than being based on the listdir results.  This is intentional since in many cases (operations on materializations with large amounts of zs) it is faster to use try/except on given zs than to compare to a listing of all available.